File: 1222-queens-that-can-attack-the-king/1222-queens-that-can-attack-the-king.js
/**
 * @param {number[][]} queens
 * @param {number[]} king
 * @return {number[][]}
 */
var queensAttacktheKing = function(queens, king) {
    queens = new Set(queens.map(([c, r]) => `${r} ${c}`))
    const res = []
    for (const r of [-1, 0, 1]) {
        for (const c of [-1, 0, 1]) {
            for (let i=1; i < 8; i++) {
                let kc = king[0] + c * i
                let kr = king[1] + r * i
                if (queens.has(`${kr} ${kc}`)) {
                    res.push([kc, kr])
                    break
                }
            }
        }
    }
    return res
};

File: 1222-queens-that-can-attack-the-king/1222-queens-that-can-attack-the-king.py
class Solution:
    def queensAttacktheKing(self, queens: List[List[int]], king: List[int]) -> List[List[int]]:
        
        queens = set([(c, r) for c, r in queens])
        res = []
        
        for r in [-1, 0, 1]:
            for c in [-1, 0, 1]:
                for i in range(1, 8):
                    kc = king[0] + c * i
                    kr = king[1] + r * i
                    
                    if (kc, kr) in queens:
                        res.append([kc, kr])
                        break
                        
        return res